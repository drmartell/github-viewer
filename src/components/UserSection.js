import React from 'react';
import { useSelector } from 'react-redux';
import { selectUserDetails, isUserLoading } from '../selectors/userSelector';
import { searchSelector, isSearching } from '../selectors/searchSelector';
import Loading from './Loading';

const UserSection = () => {
  const loading = useSelector(isUserLoading);
  const searching = useSelector(isSearching);
  const { login, name, followers, following, html_url, id } = useSelector(selectUserDetails);

  if(loading) return <Loading />;

  return !searching ? null :
    (
      <section>
        <h1>Username: {login}</h1>
        <ul>
          <li>Name: {name}</li>
          <li>Followers: {followers}</li>
          <li>Following: {following}</li>
          <li><a href={html_url}>{html_url}</a></li>
        </ul>
      </section>
    );
};

export default UserSection;
