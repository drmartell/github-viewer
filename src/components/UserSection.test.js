import React from 'react';
import { mount } from 'enzyme';
import UserSection from './UserSection';
import { Provider } from 'react-redux';
import store from '../store';

describe('UserSection component', () => {
  it('renders UserSection', () => {
    const wrapper = mount(
      <Provider store={store}>
        <UserSection />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
