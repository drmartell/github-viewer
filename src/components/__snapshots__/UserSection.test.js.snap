// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSection component renders UserSection 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <UserSection />
</Provider>
`;
